src/trackers/trackingParams.js

```diff
@@ -0,0 +1,374 @@
+// is useful to help filter out cases where a site uses the entire param string in requests and cookies. 
+// The first value in the list is the expected param, the others are altered versions that were found in manual testing.
+const falsePositiveParams = ['not_a_param', 'a_param', 'notaparam', 'not_a', 'not%20a%20param', 'not a param']
+
+// Results object will store our final param data. The format is
+// {
+//    <param> : {
+//          prevalence: percent of totalSites that the param was seen on
+//          exampleSites: list of top sites using the param in cookies or requests
+//          cookies: 
+//              firstParty: count of first party cookies this param was seen in
+//              thirdParty: count of third party cookies this param was seen in
+//              prevalence: percent of totalSites where the param was set in a cookie
+//              entities: top 10 entities using this param in cookies
+//          requests3p:
+//              prevalence: percent of totalSites where the param was seen in a third party request
+//              entities: top 10 entities using this param in requests
+//    }
+const results = {}
+let totalSites = 0
+
+function run () {
+    files.forEach(fileName => {
+        const siteData = JSON.parse(fs.readFileSync(`${config.crawlerDataLoc}/${fileName}`, 'utf8'))
+
+        if (!siteData.initialUrl) {
+            return
+        }
+
+        const siteUrl = new URL(siteData.initialUrl)
+
+        if (!siteUrl.searchParams) {
+            return
+        }
+
+        // set the global crawlParams off of the first parameter string we see
+        if (!crawlParams.paramString) {
+            setCrawlParams(siteUrl)
+        }
+
+        processRequests(siteData, siteUrl)
+        processCookies(siteData, siteUrl)
+        
+        totalSites++
+    })
+
+
+    cleanupFinalData(results)
+    fs.writeFileSync(`${config.trackerDataLoc}/build-data/generated/tracking_parameters.json`, JSON.stringify({totalSites, params: results}, null, 4))
+
+}
+run()
+
+
+// Look though the site 'call.arguments' data, count first and third party cookies
+function processCookies (siteData, siteUrl) {
+    if (!(siteData.data && siteData.data.apis && siteData.data.apis.savedCalls)) {
+        return
+    }
+
+    siteData.data.apis.savedCalls.forEach(call => {
+        
+        if (!(call.arguments && call.arguments.length)) {
+            return
+        }
+
+        if (typeof call.arguments === 'string') {
+            call.arguments = JSON.parse(call.arguments)
+        }
+
+        // find matching param and args. Don't count args that match all parameters, these are just the full param strings
+        call.arguments.forEach(arg => {
+            const argMatches = []
+            for (const [param, value] of crawlParams.params.entries()) {
+                if (arg.match(value) && !hasFalsePositiveMatch(arg)) {
```

Comment:
hm, do you really want "match" in here? It assumes "value" is a regex e.g.

> 'bla'.match('.')
< ["b"] (1)

i think you want "includes" instead?

> 'bla'.includes('.')
< false


generating those regexes from strings might be also slowign things down

Reply:
👍 you're right, thanks!